tilelang/tools/lower_trace/core.py

```
@@ -0,0 +1,1202 @@
+                before_text=before_text,
+                after_text=after_text,
+                changed=True,
+                add_lines=add_count,
+                del_lines=del_count,
+                status=STATUS_CODEGEN,
+            )
+            _records.append(record)
+            _save_raw_files(record)
+            tag = "CODEGEN"
+            path_suffix = f"  →  {_ANSI_BLUE}{codegen_out_path}{_ANSI_RESET}" if codegen_out_path else ""
+            print(f"  [lower_trace] codegen/{idx:02d}_codegen: {tag} (+{add_count}/−{del_count}){path_suffix}")
```

Ruff (0.15.18)

[warning] String contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)